src/universe/visualization/img_tools.py

```diff
@@ -14,3 +18,313 @@ def rotate(img: np.ndarray, angle: float) -> np.ndarray:
+    return cv.addWeighted(np.zeros_like(image), darkness / 100, image, 1 - darkness / 100, 0)
+
+
+def draw_lines(image: cv.Mat, lines: List[Line2d], color: Tuple[int, int, int],
+               endpoints: bool = False, line_width: float = 0.004, opacity: int = 75) -> cv.Mat:
+    """
+    Draws the given lines on the given image with the given color.
+    Make sure that the lines are scaled to the image size.
+
+    :param image: image to draw the lines on
+    :param lines: lines to draw on the image (scaled to the image size)
+    :param color: color of the lines (in the same color space as the image)
+    :param endpoints: whether to draw circles at the endpoints of the lines
+    :param line_width: this value multiplied by the width of the image is the line width in pixels
+    :param opacity: opacity of the lines (0-100)
+    :return: image with the lines drawn on it
+    """
+    assert image is not None, 'Image does not exist'
+    assert lines and len(lines) != 0, 'No lines to draw'
+    assert 0 <= opacity <= 100, 'Opacity must be between 0 and 100'
+
+    # calculate the line width in pixels based on the image width
+    line_width = round(line_width * image.shape[1])
+
+    # create an overlay for the lines
+    overlay = np.zeros_like(image)
+    for line in lines:
+        # round the points to integers
+        p1 = (round(line.a.x), round(line.a.y))
+        p2 = (round(line.b.x), round(line.b.y))
+
+        # draw line on the overlay
+        cv.line(overlay, p1, p2, color, line_width)
+        # draw endpoints on the overlay
+        if endpoints:
+            cv.circle(overlay, p1, line_width, color, -1)
+            cv.circle(overlay, p2, line_width, color, -1)
+
+    # draw the overlay on the image
+    return cv.addWeighted(overlay, opacity / 100, image, 1., 0.)
+
+
+def draw_mask(image: cv.Mat, mask: cv.Mat, color: Tuple[int, int, int], threshold: int = 30,
+              opacity: int = 75) -> cv.Mat:
+    """
+    Draws the given mask on the given image.
+
+    :param image: image to draw the mask on
+    :param mask: mask to draw
+    :param color: color of the mask (in the same color space as the image)
+    :param threshold: threshold of the mask (0-255)
+    :param opacity: opacity of the mask (0-100)
+    :return: image with the mask drawn on it
+    """
+    assert image is not None, 'Image does not exist'
+    assert mask is not None, 'Mask does not exist'
+    assert 0 <= opacity <= 100, 'Opacity must be between 0 and 100'
+
+    # resize the mask
+    mask = cv.resize(mask, (image.shape[1], image.shape[0]))
+    # convert the mask to grayscale
+    mask = cv.cvtColor(mask, cv.COLOR_RGB2GRAY)
+    # threshold the mask
+    mask[mask < threshold] = 0
+    mask[mask >= threshold] = 1
+    # create overlay
+    overlay = np.zeros_like(image)
+    # draw mask on the overlay
+    overlay[mask > 0] = color
+
+    # draw the overlay on the image
+    return cv.addWeighted(overlay, opacity / 100, image, 1., 0.)
```

Comment:
```suggestion
    return cv.addWeighted(overlay, opacity, image, 1. - opacity , 0.)
```
If 0 <= opacity <= 1.